test/extended/node_tuning/node_tuning.go

```diff
@@ -111,6 +117,81 @@ var _ = g.Describe("[sig-node-tuning] NTO should", func() {
 
 		err = fmt.Errorf("case: %v\nexpected error got because of %v", g.CurrentSpecReport().FullText(), fmt.Sprintf("stalld service restarted : %v", errWait))
 		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	//OCPBUGS-18052
+	g.It("SNO installation does not finish due to wait for non-existing machine-config [Early]", func() {
+		isSNO := isSNOCluster(oc)
+		if !isSNO {
+			g.Skip("only test on SNO cluster, skipping it ...")
+		}
+
+		var (
+			mcpConfigDaemonset *corev1.Pod
+			errMasterUpdating  error
+		)
+
+		ctx := context.TODO()
+		nodeClient := oc.KubeClient().CoreV1().Nodes()
+		firstMasterNodeName, err := getFirstMasterNode(ctx, nodeClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("assert the status of mcp master")
+
+		config, err := framework.LoadConfig()
+		framework.ExpectNoError(err)
+		dynamicClient := dynamic.NewForConfigOrDie(config)
+		//Wait for 10 minutes and check mcp master stay on normal state
+		errWait := wait.Poll(30*time.Second, 600*time.Second, func() (bool, error) {
+			mcps := dynamicClient.Resource(schema.GroupVersionResource{
+				Group:    "machineconfiguration.openshift.io",
+				Version:  "v1",
+				Resource: "machineconfigpools",
+			})
+			pools, err := mcps.List(context.Background(), metav1.ListOptions{})
+			if err != nil {
+				framework.Logf("error getting pools %v", err)
+				return false, nil
+			}
+			allUpdated := true
+			for _, p := range pools.Items {
+				updated, requiresUpdate := IsPoolUpdated(mcps, p.GetName())
+				allUpdated = allUpdated && updated
+
+				// Invariant: when CVO reaches level, MCO is required to have rolled out control plane updates
+				if p.GetName() == "master" && requiresUpdate && errMasterUpdating == nil {
+					errMasterUpdating = fmt.Errorf("the %q pool should be updated before the CVO reports available at the new version", p.GetName())
+					framework.Logf("Invariant violation detected: %s", errMasterUpdating)
+				}
+			}
+			e2e.Logf("the status of mcp master restore to normal state")
+			return allUpdated, nil
+
+		})
+
+		if errWait != nil {
+			e2e.Logf("%v", errWait)
+			return
```

Comment:
Fixed in new code, it's wrong logic in my previous code. the test case will broken if mcp keep unexpected state after 10 minutes, for wait for 10 minutes, what I want to do is  avoiding the node was reboot or mcp configuration updated by other test cases, it will caused test case failed, actually after other test case completed, the mcp will restore to updated state.